File: DSA/Sorting/bubbleSort.dart
import 'dart:io';

List bubbleSort(List list) {
  int len = list.length;
  for (int i = 0; i < len - 1; i++) {
    for (int j = 0; j < len - i - 1; j++) {
      if (list[j] > list[j + 1]) {
        //Swap the element
        int temp = list[j];
        list[j] = list[j + 1];
        list[j + 1] = temp;
      }
    }
  }
  return list;
}

void main() {
  print("Enter the size of list");
  int size = int.parse(stdin.readLineSync()!);
  List list = List.generate(size, (generator) {
    print("Enter the element: ");
    int value = int.parse(stdin.readLineSync()!);
    return value;
  });
  print("Original List is: $list");
  var bubble_sort = bubbleSort(list);
  print(bubble_sort);
}

File: DSA/Python/Sorting/bubbleSort.py
def bubbleSort(list):
    length = len(list)
    for i in range(length - 1):
        for j in range(length - 1 - i):
            if list[j] > list[j + 1]:
                list[j], list[j + 1] = list[j + 1], list[j]
    return list

size = int(input("Enter the size of the list: "))
element = [int(input("Enter the elements: ")) for i in range(size)]

bubSort = bubbleSort(element)
print(bubSort)
